### PR TITLE
fix: Raise conflict exception on creating attendee on ticket which is sold out.

### LINKED ISSUE #4961: ConflictException wrongly raised in the before_post of Attendee
**Describe the bug**

When creating an attendee ConflictException is raised when we try to create an attendee on a ticket which already has an attendee. It is not correct. 
If we have a ticket whose quantity is 10, it will not allow creating more than one attendee on that ticket even though it should allow upto 10 attendees on it.

**To Reproduce**
1. Try to create an attendee on a ticket which already has an Attendee.

**Expected behavior**

It should check for number of TicketHolders with that ticket and compare it with the ticket quantity.

**Additional context**

Related code:
```Python
        if db.session.query(TicketHolder.id).filter_by(
                ticket_id=int(data['ticket']), event_id=int(data['event'])
                ).scalar() is not None:
            raise ConflictException(
                {'pointer': '/data/attributes/ticket_id'},
                "Attendee with this ticket already exists for the same event"
            )
```

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #4961 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
When creating an attendee ConflictException is raised when we try to create an attendee on a ticket which already has an attendee. It is not correct.
If we have a ticket whose quantity is 10, it will not allow creating more than one attendee on that ticket even though it should allow upto 10 attendees on it.

#### Changes proposed in this pull request:
Raise conflict exception on creating attendee on ticket which is sold out.

a
b

### CODE REVIEW

File: app/api/attendees.py [hunk 1]
```diff
@@ -47,13 +47,12 @@ def before_post(self, args, kwargs, data):
...
-        if db.session.query(TicketHolder.id).filter_by(
-                ticket_id=int(data['ticket']), event_id=int(data['event'])
-                ).scalar() is not None:
+        # Check if the ticket is already sold out or not.
+        if get_count(db.session.query(TicketHolder.id).
+                     filter_by(ticket_id=int(data['ticket']), deleted_at=None)) >= ticket.quantity:
             raise ConflictException(
                 {'pointer': '/data/attributes/ticket_id'},
-                "Attendee with this ticket already exists for the same event"
+                "Ticket already sold out"
```
Comment (schedutron) [inline app/api/attendees.py]: Could there be a better message? Like _all such_ tickets have been sold out?

Comment (Kreijstal) [reply app/api/attendees.py]: Does ticket id refer to just 1 ticket, or to the tickets in general for an event.

Comment (srv-twry) [reply app/api/attendees.py]: Tickets in general. 
for eg. Ticket has a field quantity. So if the quantity is 100, then 100 people can buy that ticket.

Comment (mayank8318) [reply app/api/attendees.py]: Or maybe just "Sold out"

**Comment by bhaveshAn** [comment]
Please resolve conflicts

**Comment by srv-twry** [comment]
@bhaveshAn Done. Please review.

**Pull request merged**